### PR TITLE
pass correlation id to apis for moc

### PR DESCRIPTION
**Pull request by lpdnju**
adding correlation id in ctx that used to call moc api.
related changes in moc: https://github.com/microsoft/moc/pull/203/files

### CODE REVIEW

File: cloud/scope/cluster.go [hunk 1]
```diff
@@ -71,14 +72,16 @@ func NewClusterScope(params ClusterScopeParams) (*ClusterScope, error) {
...
+	scopeContext := diagnostics.NewContextWithCorrelationId(context.Background(), params.AzureStackHCICluster.GetAnnotations()[infrav1.AzureCorrelationIDAnnotationKey])
 	scope := &ClusterScope{
 		Logger:               *params.Logger,
 		Client:               params.Client,
 		AzureStackHCIClients: params.AzureStackHCIClients,
 		Cluster:              params.Cluster,
 		AzureStackHCICluster: params.AzureStackHCICluster,
 		patchHelper:          helper,
-		Context:              context.Background(),
+		Context:              scopeContext,
```
Comment (pradipd) [inline cloud/scope/cluster.go:84]: So how does the correlation id get sent to moc?
i see that this is adding a new context w/ correclation id to the scope, but, I'm not understanding how moc will use it.

Comment (pradipd) [reply cloud/scope/cluster.go:84]: is there a corresponding change in wssdcloudagent/wssdagent ?

**Review by pradipd — APPROVED** [review]

**Comment by pradipd** [comment]
I have approved. Semantically, the changes look fine.   And we can start getting this into cloud-op.

But, I'm still don't understand how moc stack will consume this.  If there are relevant moc PR's please let me know.

File: cloud/scope/cluster.go (hunk 1 above)
Comment (lpdnju) [reply cloud/scope/cluster.go:84]: Yes, changes:
https://github.com/microsoft/moc-pkg/pull/420/files
https://github.com/microsoft/wssdcloudagent/pull/802
https://github.com/microsoft/wssdagent/pull/797

Comment (pradipd) [reply cloud/scope/cluster.go:84]: ah. now i see.  Thanks!

**Pull request merged**